include/pybind11_json/pybind11_json.hpp

```diff
@@ -129,4 +129,48 @@ namespace nlohmann
 
 }
 
+namespace pybind11
+{
+    namespace detail
+    {
+        template <> struct type_caster<nlohmann::json>
+        {
+        public:
+            /**
+             * This macro establishes the name 'user_type' in
+             * function signatures and declares a local variable 'value' of user type
+             */
+            PYBIND11_TYPE_CASTER(nlohmann::json, _("json"));
+
+            /**
+             * Conversion part 1 (Python->C++): convert a PyObject into a inty
+             * instance or return false upon failure. The second argument
+             * indicates whether implicit conversions should be applied.
+             */
+            bool load(handle src, bool)
+            {
+                /* Extract PyObject from handle */
+                /* Now try to convert into a C++ type */
+                value = nlohmann::detail::to_json_impl(src);
+                if (value.is_null())
+                    return false;
+                return true;
+            }
+
+            /**
+             * Conversion part 2 (C++ -> Python): convert an inty instance into
```

Comment:
I think `inty` comes from the PYBIND11 documentation here, you might want to replace it with `nlohmann::json`. 

I would be fine removing all the comments though :)